stl/inc/expected

```diff
@@ -685,13 +700,393 @@ public:
         }
     }
 
-    // [expected.object.eq]
+    template <class _Uty = _Err>

```

Comment:
No changes requested; it's odd to me that `_Uty` is defaulted here, but not for `value_or`; is this worth an LWG issue?

Reply:
From the LEWG notes, the default template argument here is to support migration from error handling patterns like:

```cpp
error_code ec = produce(&t);
if (test_for_error(ec)) 
  do_something();
```

Which could become:

```cpp
expected<T, error_code> e = produce();
if (test_for_error(e.error_or({}))
   do something();
```

I don't see any discussion in the notes about API symmetry with `value_or` for this. I guess they saw a use case for `error_or` to allow that `{}` argument but not so much for `value_or`.

Reply:
Consensus on the reflector seems to be that the template type parameter for `value_or` should be defaulted to `remove_cv_t<T>` for both `expected` and `optional`. Has anyone submitted an LWG issue to do so, or should I?

Reply:
@CaseyCarter go ahead!

Reply:
Submitted an issue to default the template argument to `remove_cv_t<T>` for the forwarding-reference constructor, assignment operator, and both `value_or` overloads of `expected` and `optional`.